tests/test_browser.py

```diff
@@ -77,6 +78,7 @@ def test_build_request():
     request = browser._prepare_request(form)
     assert "application/x-www-form-urlencoded" in request.headers[
         "Content-Type"]
+    browser.close()
```

Comment:
I'm confused about all these calls to `close()`. Isn't this called when `browser` is destructed (due to going out of function scope)?

Reply:
I don't fully understand the issue, but the issue does occur: http://mechanicalsoup.readthedocs.io/en/latest/faq.html#referenceerror-weakly-referenced-object-no-longer-exists

The warning is essentially harmless, I saw a lot of them with tests passing (i.e. `python setup.py test` showed the warning and still concluded that the tests were passing) before I added the `.close` calls. But some of the tests check stderr, and these tests were flaky. See e.g. https://travis-ci.org/MechanicalSoup/MechanicalSoup/builds/290550991 for a failure.

More info here:
https://github.com/requests/requests/issues/2303
Essentially the conclusion is "don't do anything complex in `__del__`" and it seems `session.close()` is already too complex. But I may well have missed something.